Firmware/MotorControl/encoder.cpp

```diff
@@ -678,7 +678,7 @@ bool Encoder::update() {
                                 edge_idx = hall_cnt;
                             } else if (mod_hall_cnt == 5) { // counted down
                                 edge_idx = last_hall_cnt_.value();
-                            } else {
+                            } else if(!config_.ignore_illegal_hall_state){
```

Comment:
This will lead to `edge_idx` being uninitialized when it's used further down the function.